x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportOpenJobAction.java

```diff
@@ -171,51 +171,49 @@ public void onFailure(Exception e) {
             );
 
             // Tell the job tracker to refresh the memory requirement for this job and all other jobs that have persistent tasks
-            ActionListener<Boolean> referencedRuleFiltersPresentListener = ActionListener.wrap(
-                response -> memoryTracker.refreshAnomalyDetectorJobMemoryAndAllOthers(
-                    jobParams.getJobId(),
-                    memoryRequirementRefreshListener
-                ),
-                listener::onFailure
+            ActionListener<Boolean> referencedRuleFiltersPresentListener = memoryRequirementRefreshListener.delegateFailureAndWrap(
+                (l, response) -> memoryTracker.refreshAnomalyDetectorJobMemoryAndAllOthers(jobParams.getJobId(), l)
             );
 
             // Validate referenced rule filters are present
-            ActionListener<Boolean> modelSnapshotValidationListener = ActionListener.wrap(response -> {
-                Set<String> referencedRuleFilters = jobParams.getJob().getAnalysisConfig().extractReferencedFilters();
-                if (referencedRuleFilters.isEmpty()) {
-                    referencedRuleFiltersPresentListener.onResponse(true);
-                } else {
-                    GetFiltersAction.Request getFiltersRequest = new GetFiltersAction.Request();
-                    getFiltersRequest.setResourceId(String.join(",", referencedRuleFilters));
-                    getFiltersRequest.setAllowNoResources(false);
-                    client.execute(
-                        GetFiltersAction.INSTANCE,
-                        getFiltersRequest,
-                        ActionListener.wrap(filtersResponse -> referencedRuleFiltersPresentListener.onResponse(true), listener::onFailure)
```

Comment:
These chains of wrapping are the only non-trivial change in this PR. E.g. in this one we move the failure handling from `listener::onFailure` to `referencedRuleFiltersPresentListener::onFailure` technically speaking. This is a non-change though because referencedRuleFiltersPresentListener.onFailure delegates to `listener.onFailure` and the same holds true for subsequent wrapping use.